alacritty/src/config/ui_config.rs

```diff
@@ -169,3 +195,141 @@ pub struct Delta<T: Default> {
     /// Vertical change.
     pub y: T,
 }
+
+/// Regex terminal hints.
+#[derive(ConfigDeserialize, Default, Debug, PartialEq, Eq)]
+pub struct Hints {
+    /// Characters for the hint labels.
+    alphabet: HintsAlphabet,
+
+    /// Available hint configurations.
+    enabled: Vec<Hint>,
+}
+
+impl Hints {
+    /// Characters for the hint labels.
+    pub fn alphabet(&self) -> &str {
+        &self.alphabet.0
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct HintsAlphabet(String);
+
+impl Default for HintsAlphabet {
+    fn default() -> Self {
+        Self(String::from("jfkdls;ahgurieowpq"))
+    }
+}
+
+impl<'de> Deserialize<'de> for HintsAlphabet {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
+
+        let mut character_count = 0;
+        for character in value.chars() {
+            if character.width() != Some(1) {
+                return Err(D::Error::custom("characters must be of width 1"));
+            }
+            character_count += 1;
+        }
+
+        if character_count < 2 {
+            return Err(D::Error::custom("must include at last 2 characters"));
```

Comment:
This is not correct check, you need 2 **unique** characters, not just 2 characters, right now you allow alphabet like `aa`, however it's not really usable, so you could build a `Set<char>`(and carry it around) or remove all repeating chars and then check.

Reply:
It's not usable, but it works perfectly fine. I've considered having more checks like unique characters but ultimately that just leads to increased startup time without benefit to most people.

The problem with just a single character is that things will straight-up fail. But with the same character X times the user should figure out the error themselves, since it is clearly visible. But Alacritty still manages to work just fine, it just doesn't make any sense to do so.